app/src/main/java/com/base/firebase_operations/ImageAdapter.kt

```diff
@@ -27,6 +35,29 @@ class ImageAdapter(val context: Context, val uri:List<String>):RecyclerView.Adap
     override fun onBindViewHolder(holder: imageviewholder, position: Int) {
         val currenturl=uri[position]
         Glide.with(context).load(currenturl).into(holder.imageholder)
+        holder.imageholder.setOnClickListener {
```

Comment:
In this code block we just set a onclick listener on the images present inside the recycler view and created a reference variable which would help us in getting the names of the images present in our firebase storage.Now the thing is that we have uploaded the images using the system.currenttimeinmills as a child parameter meaning that every image that would be uploaded to the cloud storage would be having a unique timestamp,and the timestamp or that particular value would be acting as a identifier for the images and so extract that particular part from the whole name we just implemented a small logic in which we converted the obtained name into a character array and define the startindex and endindex for getting the desired characters and used the copyofRange function for copying those characters inside a new array and here inside the copyofRange function we passed the startindex and endindex includes a +1 which means that the character at 21 index is included in the copied array.
After getting that unique identifier we just created a coroutine and executed the delete function and here the child would be containing the unique identifier name and would be deleting only that image that is been clicked from the Firebase storage.